src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7525,28 +7571,14 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
             }
 
             // An "embedded" statement is simply a statement that is not a labelled
-            // statement or a declaration statement.  Parse a normal statement and post-
-            // check for the error case.
+            // statement or a declaration statement.
             switch (statement.Kind)
             {
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
-                case SyntaxKind.ExpressionStatement:
-                    // Deconstruction-declaration is only allowed as top-level statement
-                    // see https://github.com/dotnet/roslyn/issues/15049
-                    var expression = ((ExpressionStatementSyntax)statement).Expression;
-                    if (expression.Kind == SyntaxKind.SimpleAssignmentExpression)
-                    {
-                        var assignment = (AssignmentExpressionSyntax)expression;
-                        if (assignment.Left.IsDeconstructionDeclarationLeft())
-                        {
-                            statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
```

Comment:
Yes, the LDM explicitly decided to remove the language concept of a deconstruction declaration statement, which is where this restriction arises. We will discuss tomorrow if the LDM wants to add back any of these restrictions in C# 7, including the ones among the first changes in `Binder_Deconstruct.cs` (lines 26-57 in this PR); as of the last LDM they wanted to remove all those restrictions. But when we do we will need additional tests.